stdlib/SparseArrays/src/sparsematrix.jl

```diff
@@ -376,6 +376,27 @@ function SparseMatrixCSC{Tv,Ti}(M::AbstractMatrix) where {Tv,Ti}
     eltypeTvV = convert(Vector{Tv}, V)
     return sparse_IJ_sorted!(eltypeTiI, eltypeTiJ, eltypeTvV, size(M)...)
 end
+function SparseMatrixCSC{Tv,Ti}(M::Matrix) where {Tv,Ti}
+    nz = count(!equalto(0), M)
```

Comment:
This count is needed to avoid using `push!` which unfortunately is slow due to its overhead (https://github.com/JuliaLang/julia/issues/24909)

Reply:
Don't you want `!equalto(zero(Tv))` here? Thinking about matrices with SVector entries.

Reply:
#24790